packages/calcite-components/src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -192,13 +290,27 @@ export class DatePickerMonthHeader {
 
   @Element() el: HTMLCalciteDatePickerMonthHeaderElement;
 
-  private yearInput: HTMLInputElement;
+  @State() nextMonthDate: Date;
+
+  @State() prevMonthDate: Date;
+
+  @State() yearList: number[] = [];
 
   private parentDatePickerEl: HTMLCalciteDatePickerElement;
 
-  @State() nextMonthDate: Date;
+  private yearPickerEl: HTMLCalciteSelectElement;
 
-  @State() prevMonthDate: Date;
+  private monthPickerEl: HTMLCalciteSelectElement;
+
+  private defaultMinYear = 1950;
+
+  private defaultMaxYear = 2050;
```

Comment:
Could the min/max years be derived from the min/max ISO year vars?

Reply:
pending Discussion.